### DISCUSSION TITLE
Taxonomy Search Zero Results in Search Filters, but works with Querystring

### DISCUSSION THREAD

**Discussion by josy1024** [microsoft-search/pnp-modern-search#1694] [Q&A]
My Search Refiner Filter on Metadata Field ist not working as expected (V4) ;-(

Environment: 
* Search box from Query string (q=)
* Search Refiner with Metadata Field. (currently with 2. Filter RefinableString00, but does not matter)
* Search Results Box Cross-Connected to Refiner

Tags Patched to use Query String Parameter "Q"
```
{{#each (split (slot item @root.slots.Tags) ',') as |tag| }}
			<a style="margin-left:2px" href="?q=owstaxidmetadataalltagsinfo:'{{trim tag}}'">{{tag}}</a>
```

querystring search for: owstaxidmetadataalltagsinfo:'Logos' Gets 4 Results.
![image](https://user-images.githubusercontent.com/7140712/148796113-d87a35e1-a46d-407a-8095-bd3464ef701f.png)

search * gets all (9 results) => shows up 4 results on logos.
Termstore search for "logos" gets zero results??

![image](https://user-images.githubusercontent.com/7140712/148796209-3a93f383-f9c7-4c71-a4ec-6adb5245d4ca.png)

Metadata shows up correctly.
![image](https://user-images.githubusercontent.com/7140712/148797201-b3e7d992-c483-456f-afd9-e3eb40ce0573.png)

any ideas? @wobba

**Comment by wobba** [comment]
Without knowing how your metadata field is set up in SharePoint, and if you have mapped some of the crawled properties for it - it's hard to  say exactly why you are seeing the behavior you see. You could for example be hit by https://www.techmikael.com/2020/03/side-effect-of-mapping-crawled.html

**Reply by josy1024** [reply]
thanks for the tip! hm.. - i've tried in 2 different tenants, different metadata-sets

What i'v done:
in Filter:
Add Custom Filter: Tag=owstaxidmetadataalltagsinfo
![image](https://user-images.githubusercontent.com/7140712/149084521-3cb19f5a-194a-4972-a584-47b01c407d6b.png)

In Search Result:
Add Property owstaxidmetadataalltagsinfo 

Layout Slots:
Tag owstaxidmetadataalltagsinfo Change to AUTOowstaxidmetadataalltagsinfo
like mentioned in: https://github.com/microsoft-search/pnp-modern-search/issues/85#issuecomment-814951554

what i'v not done: map the metadata to a RefinableStringXX Field? - is that required for V4?


Search and Filtering is working with anther RefinableStringXX from a Lookup Column.

V3 Metadata Search with Taxonomy works as expected, so i don't think its of a generel Search/Index Problem.
I did not have to map the Termstore in V3 modern search.

Klicking on the Filter Zeros Results.

![image](https://user-images.githubusercontent.com/7140712/149086246-98b5025f-41d4-4d9d-b609-f2364fdbfb2c.png)

@wobba am i missing something?

**Reply by FranckyC** [reply]
Hi @josy1024, seems related to https://github.com/microsoft-search/pnp-modern-search/issues/1702. Can you try using a RefinableStringXX property as a workaround?

**Reply by josy1024** [reply]
THANKS! @FranckyC  - had success with a RefinabelStringXX Workaround. Looks definitly like this issu: https://github.com/microsoft-search/pnp-modern-search/issues/1702 (fix: https://github.com/microsoft-search/pnp-modern-search/pull/1703)

some Term and RefinableStringXX  related links...
* Full Path to Term in the Search Results Handlebars Template https://github.com/microsoft-search/pnp-modern-search/discussions/1360
* Autor mapping filter the localSharePointresult using the CreatedBy or author fields https://github.com/microsoft-search/pnp-modern-search/discussions/1555
* Autoowstaxidmetadataalltagsinfo https://github.com/microsoft-search/pnp-modern-search/issues/1301

**Comment by josy1024** [comment]
you can verify ALL RefinableString mappings with:

TENANT:
```powershell
Connect-PnPOnline -Url "https://#TENANT#-admin.sharepoint.com" -Interactive
Get-PnPSearchConfiguration -Scope Subscription -OutputFormat ManagedPropertyMappings
```

SITE:
```powershell
Connect-PnPOnline -Url "https://#TENANT#.sharepoint.com/sites/#SITE#" -Interactive
Get-PnPSearchConfiguration -Scope Site -OutputFormat ManagedPropertyMappings
```
